Donation-platform/src/main/java/com/google/servlets/PersonalizationServlet.java

```diff
@@ -29,29 +31,37 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns personalized comments data.*/
+
+/* Servlet that returns personalized charities according to the user's selected causes.*/
 @WebServlet("/personalize")
 public class PersonalizationServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     List<String> tags = new ArrayList<>();
 
-    // Get the tag input from the checkbox form
-    String blm = request.getParameter("blm");
-    String education = request.getParameter("education");
-    String hungerAndPoverty = request.getParameter("hunger-and-poverty");
-    String environment = request.getParameter("environment");
+    // Get the user's tag input from the checkbox form
+    String tag1 = request.getParameter("tag-1");
+    String tag2 = request.getParameter("tag-2");
+    String tag3 = request.getParameter("tag-3");
```

Comment:
What if the user only enters 2 tags?

Reply:
Just tested that and it still works properly (doesn't matter which of the fields is the one left blank). It also works if you only enter one tag. If you enter no tags, it returns an empty list (which for now is an adequate response, in my opinion).